preload/preload-resource-match.https.sub.html

```diff
@@ -0,0 +1,156 @@
+        div.style.shapeOutside = `url(${href})`;
+        document.body.appendChild(div);
+        t.add_cleanup(() => div.remove());
+    },
+    backgroundImage: (href, attr, t) => {
+        const div = document.createElement('div');
+        div.style.background = `url(${href})`;
+        document.body.appendChild(div);
+        t.add_cleanup(() => div.remove());
+    },
+    fetch: async (href, attr, t) => {
+        const options = {mode: attr.crossOrigin ? 'cors' : 'no-cors',
+             credentials: !attr.crossOrigin || attr.crossOrigin === 'anonymous' ? 'omit' : 'include'}
+
+        const response = await fetch(href, options)
+        await response.text();
+    },
+    script: async (href, attr, t) => {
+        const script = document.createElement('script');
+        t.add_cleanup(() => script.remove());
+        if (attr.crossOrigin)
+            script.setAttribute('crossorigin', attr.crossOrigin);
+        script.src = href;
+        document.body.appendChild(script);
+        await new Promise(resolve => { script.onload = resolve });
+    },
+    module: async (href, attr, t) => {
+        const script = document.createElement('script');
+        script.type = 'module';
+        t.add_cleanup(() => script.remove());
+        if (attr.crossOrigin)
+            script.setAttribute('crossorigin', attr.crossOrigin);
+        script.src = href;
+        document.body.appendChild(script);
+        await new Promise(resolve => { script.onload = resolve });
+    },
+    style: (href, attr, t) => {
+        const style = document.createElement('link');
+        style.rel = 'stylesheet';
+        t.add_cleanup(() => style.remove());
+        if (attr.crossOrigin)
+            style.setAttribute('crossorigin', attr.crossOrigin);
+        document.body.appendChild(style);
+    }
+}
+
+function preload_reuse_test(type, as, url, preloadConfig, resourceConfig) {
+    const expected = (preloadConfig === resourceConfig) ? "reuse" : "discard";
+    const key = token();
+    const href = getAbsoluteURL(`${
+        (configs[resourceConfig].crossOrigin ? crossOriginHost : '') + url
+    }&${token()}`)
+    promise_test(async t => {
+        await preload({href, as, ...configs[preloadConfig].attributes}, t);
+        await loaders[as](href, configs[resourceConfig].attributes, t)
+        verifyNumberOfResourceTimingEntries(href, expected === "reuse" ? 1 : 2)
+    }, `Loading ${type} (${resourceConfig}) with link (${preloadConfig}) should ${expected} the preloaded response`);
+}
+
+for (const resourceTypeName in resourceTypes) {
+    const resourceInfo = resourceTypes[resourceTypeName];
+    const configNames = resourceInfo.config ? [resourceInfo.config, 'same-origin'] : Object.keys(configs)
+    for (const resourceConfigName of configNames) {
+        for (const preloadConfigName of Object.keys(configs)) {
+            if (resourceConfigName === 'same-origin' && preloadConfigName !== 'same-origin')
+                continue;
```

Comment:
Nit: `||` those 2 conditions? (instead of having those separate `if` statements)

Reply:
Also, we're skipping those cases because in those cases we'd get a different URL?

Reply:
Because if it's the same origin the CORS attributes are ignored anyway. Will try to clarify with a comment